frontend/src/main.tsx

```diff
@@ -4,19 +4,37 @@ import { App } from "./App.tsx";
 import "./index.css";
 import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
 import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
+import { RouterProvider, createRouter } from "@tanstack/react-router";
+
+import { routeTree } from "./routeTree.gen";
+const router = createRouter({ routeTree });
+
+// Register the router instance for type safety
+declare module "@tanstack/react-router" {
+  interface Register {
+    router: typeof router;
+  }
+}
+
```

Comment:
put this inside of a `.d.ts` file (ex: `tanstack-router.d.ts` and include that file in `tsconfig.json`.
```suggestion
import {} from "@tanstack/react-router";

// Register the router instance for type safety
declare module "@tanstack/react-router" {
  interface Register {
    router: typeof router;
  }
}
```

In `tsconfig.json` :
```
 "include": ["src", "./src/tanstack-router.d.ts"],
```